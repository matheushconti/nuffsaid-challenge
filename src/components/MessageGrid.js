import React from 'react'
import { Grid, Typography, Link, makeStyles } from '@material-ui/core'
import Message from './Message'

const useStyles = makeStyles((theme) => ({
  titleContainer: {
    marginBottom: 15
  },
  title: {
    fontWeight: 'bold'
  }
}))

export default function MessageGrid(props) {
  const classes = useStyles()
  const { messages, type, setMessages } = props

  const removeMessage = (k) => {
    let newMessage = messages.filter((item, key) => key !== k)
    setMessages(newMessage)
  }

  return (
      <Grid item xs={4}>
        <Grid className={classes.titleContainer}>
          <Typography variant="h5" className={classes.title}>
            {type.title}
          </Typography>
          <Typography variant="body2">
            {`Count ${messages.length}`}
          </Typography>
        </Grid>
        <Grid>
          {messages && messages.map((message,key) => 
            <Message 
              key={key} 
              datakey={key} 
              message={message}
              type={type}
              removeMessage={removeMessage}
            />
          )}
        </Grid>
      </Grid>
  )

}
