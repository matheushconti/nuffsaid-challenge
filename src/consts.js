export const messageTypes = {
    error: {priority: 1, color: '#F56236', title:'Error Type 1'},
    warning: {priority: 2, color: '#FCE788', title:'Warning Type 2'},
    info: {type: 3, color: '#88FCA3', title:'Info Type 3'}
}