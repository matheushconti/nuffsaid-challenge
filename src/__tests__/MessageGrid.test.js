import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import MessageGrid from '../components/MessageGrid'
import { messageTypes } from '../consts'

configure({ adapter: new Adapter() });

const messagesError = [{priority: 1, message: "error message"}];
const setMessagesError = jest.fn();
const messagesWarning = [{priority: 2, message: "warning message"}];
const setMessagesWarning = jest.fn();
const messagesInfo = [{priority: 3, message: "info message"}];
const setMessagesInfo = jest.fn();

describe('Testing MessageGrid component', () => {
    test('renders grid for priority 1 (error)', () => {
        const tree = renderer.create(
            <MessageGrid messages={messagesError} setMessages={setMessagesError} type={messageTypes.error} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('renders grid for priority 2 (warning)', () => {
        const tree = renderer.create(
            <MessageGrid messages={messagesWarning} setMessages={setMessagesWarning} type={messageTypes.warning} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('renders grid for priority 3 (info)', () => {
        const tree = renderer.create(
            <MessageGrid messages={messagesInfo} setMessages={setMessagesInfo} type={messageTypes.info} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("new error message", () => {
        const wrapper = shallow(<MessageGrid messages={messagesError} setMessages={setMessagesError} type={messageTypes.error} />);
        const newMessages = [{priority: 1, message: "error message"}, {priority: 1, message: "newest error message"}];
        wrapper.setProps({messages: newMessages});
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("new warning message", () => {
        const wrapper = shallow(<MessageGrid messages={messagesWarning} setMessages={setMessagesWarning} type={messageTypes.warning} />);
        const newMessages = [{priority: 2, message: "warning message"}, {priority: 2, message: "newest warning message"}];
        wrapper.setProps({messages: newMessages});
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("new info message", () => {
        const wrapper = shallow(<MessageGrid messages={messagesInfo} setMessages={setMessagesInfo} type={messageTypes.info} />);
        const newMessages = [{priority: 2, message: "info message"}, {priority: 2, message: "newest info message"}];
        wrapper.setProps({messages: newMessages});
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    })
})