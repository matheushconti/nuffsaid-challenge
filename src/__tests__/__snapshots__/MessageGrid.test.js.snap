// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MessageGrid component new error message 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Error Type 1
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 2
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#F56236",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            error message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#F56236",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            newest error message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing MessageGrid component new info message 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Info Type 3
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 2
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#88FCA3",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            info message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#88FCA3",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            newest info message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing MessageGrid component new warning message 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Warning Type 2
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 2
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#FCE788",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            warning message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#FCE788",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            newest warning message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing MessageGrid component renders grid for priority 1 (error) 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Error Type 1
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 1
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#F56236",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            error message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing MessageGrid component renders grid for priority 2 (warning) 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Warning Type 2
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 1
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#FCE788",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            warning message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing MessageGrid component renders grid for priority 3 (info) 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
>
  <div
    className="MuiGrid-root makeStyles-titleContainer-1"
  >
    <h5
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
    >
      Info Type 3
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Count 1
    </p>
  </div>
  <div
    className="MuiGrid-root"
  >
    <div
      className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      style={
        Object {
          "backgroundColor": "#88FCA3",
        }
      }
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
      >
        <div
          className="MuiGrid-root makeStyles-messageContainer-3 MuiGrid-item MuiGrid-grid-sm-true"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            info message
          </p>
        </div>
        <div
          className="MuiGrid-root makeStyles-clearContainer-4 MuiGrid-item"
        >
          <p
            className="MuiTypography-root MuiTypography-body2"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Clear
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
